guardrails/schema/string_schema.py

```diff
@@ -136,6 +136,9 @@ def validate(
         disable_tracer: Optional[bool] = True,
         **kwargs,
     ) -> Any:
+        # TODO: add class field to track number of chunks accumulated
```

Comment:
I think the class field should not be the number of chunks accumulated. Instead we should introduce a few class level items
1. a field that contains a string that's awaiting validation. WE can call this something like accumulated_chunks
2. a function that 
  a. chunks - splits the accumulated string into validatable-chunks. Returns all chunks or an empty list
  b. remove returned chunks from accumulated_chunks

Reply:
@zsimjee the functionality you describe above makes more sense to live in the Validator imo.  Especially since Schema will be going away very soons, I'd shy away from adding any important logic in here and let it all be passthrough from the Runner to the Validator.